systems-state-manager/nisystemsstate.yml

```diff
@@ -0,0 +1,999 @@
+      parameters:
+        - name: id
+          in: path
+          description: ID of state that needs to be updated.
+          required: true
+          type: string
+        - in: body
+          name: patch
+          required: true
+          description: Json Patch describing the changes.
+          schema:
+            $ref: '#/definitions/StatePatchRequest'
+      responses:
+        200:
+          description: OK
+          schema:
+            $ref: '#/definitions/StateResponse'
+        400:
+          $ref: '#/responses/BadRequest'
+        401:
+          $ref: '#/responses/Unauthorized'
+        404:
+          $ref: '#/responses/NotFound'
+        409:
+          $ref: '#/responses/Conflict'
+        default:
+          $ref: '#/responses/Error'
+    delete:
+      operationId: DeleteState
+      x-ni-privilege: nisystemsstate.Write
+      x-ni-operation: deleteStates
+      tags: [states]
+      summary: Delete state
+      description: Delete a state by providing its id.
+      parameters:
+        - name: id
+          in: path
+          description: ID of state that needs to be deleted.
+          required: true
+          type: string
+      responses:
+        204:
+          description: No Content
+        401:
+          $ref: '#/responses/Unauthorized'
+        404:
+          $ref: '#/responses/NotFound'
+        default:
+          $ref: '#/responses/Error'
+  /v1/replace-state-content:
+    post:
+      operationId: ReplaceContent
+      x-ni-privilege: nisystemsstate.Write
+      x-ni-operation: createOrUpdateStates
+      tags: [states]
+      summary: Replace the content of a state
+      description: Replace the content of a state by providing the new content of the state.
+      consumes:
+        - multipart/form-data
+      parameters:
+        - name: id
+          in: formData
+          description: ID of the state for which to update the content.
+          required: true
+          type: string
+        - in: formData
+          description: New content of the state.
+          name: file
+          type: file
+          required: true
+        - in: formData
+          description: Description of the change operation.
+          name: changeDescription
+          type: string
+      responses:
+        200:
+          description: OK
+          schema:
+            $ref: '#/definitions/StateResponse'
+        400:
+          $ref: '#/responses/BadRequest'
+        401:
+          $ref: '#/responses/Unauthorized'
+        409:
+          $ref: '#/responses/Conflict'
+        default:
+          $ref: '#/responses/Error'
+  /v1/states/{id}/history:
+    get:
+      operationId: GetStateHistory
+      x-ni-privilege: Read
+      x-ni-operation: getStates
+      tags: [history]
+      summary: Get state history
+      description: Returns a list of (version, description) pairs defining the history of the state.
```

Comment:
Would it be okay to say "version-description pairs" instead? Are they like "key-value pairs" in that sense?